### PR TITLE
test(flink): enable turning on local test env

### PR DESCRIPTION
**Pull request by mfatihaktas**
## Description of changes

Aims to unblock people who cannot run the tests with Flink backend on the dockerized env as documented in https://github.com/ibis-project/ibis/issues/8507.

Allows running the tests with local env when env variable `IBIS_FLINK_LOCAL_ENV` is set to `true` or `1`.
E.g.
```
$ IBIS_FLINK_LOCAL_ENV=1 pytest -m flink ...

$ IBIS_FLINK_LOCAL_ENV="True" pytest -m flink ...

$ IBIS_FLINK_LOCAL_ENV=true pytest -m flink ...
```

### CODE REVIEW

**Comment by cpcloud** [comment]
I'd rather we wait until we can fix this without introducing external state like environment variables to make stuff work. Is there a particular reason we can't wait for the fix?

**Comment by mfatihaktas** [comment]
> I'd rather we wait until we can fix this without introducing external state like environment variables to make stuff work. Is there a particular reason we can't wait for the fix?

Sure, I created this PR as the change is straightforward and quick. During today's triage meeting, we talked about people having issues time-to-time with setting up the Flink env on M1/M2. Even if we do not merge this PR, it can be used as a reference when needed by people who need the local Flink env to unblock themselves.

**Comment by mfatihaktas** [comment]
@cpcloud Actually, even without the env variable, I think just the part with `get_table_env()` is worth to merge because it brings visibility into which env is local and which env is remote, and making it easy to change if/when needed.

**Comment by kszucs** [comment]
I am in favor of this change. There are various environments where the docker setup may not be feasible or not working out of the box and with little code change  the developer can fall back to the local env.

File: ibis/backends/flink/tests/conftest.py
Comment (cpcloud) [inline ibis/backends/flink/tests/conftest.py:79]: When would we ever set this to `false`?

**Comment by cpcloud** [comment]
Is this PR still necessary?

File: ibis/backends/flink/tests/conftest.py
Comment (mfatihaktas) [reply ibis/backends/flink/tests/conftest.py:79]: I guess you are referring to the fact that `IBIS_FLINK_LOCAL_ENV` is binary, and we can simply interpret it being defined as the flag to run the tests locally. I chose to make it explicit with `true` and `false`, mostly due to personal preference over possibly having to `unset` env variables.

**Comment by mfatihaktas** [comment]
> Is this PR still necessary?

It is not necessary to keep `IBIS_FLINK_LOCAL_ENV `. But I think there is value in keeping `get_table_env()` per my previous comment:

> @cpcloud Actually, even without the env variable, I think just the part with get_table_env() is worth to merge because it brings visibility into which env is local and which env is remote, and making it easy to change if/when needed.

File: ibis/backends/flink/tests/conftest.py
Comment (cpcloud) [reply ibis/backends/flink/tests/conftest.py:79]: Sounds good, let's just delete the variable then :)

**Review by cpcloud — APPROVED** [review]
LGTM, modulo removal of the environment variable.

**Comment by cpcloud** [comment]
@mfatihaktas Needs a rebase.

**Pull request merged**
